README.md
# X-MU_Community_Server
Всем привет, это открытый проэкт сервера му-комьюнити форума х-му.


zGameServer_X/ChangeLog.txt
// ======================================================================
// Project:			zGameServer
// ----------------------------------------------------------------------
// Version:			Season 6 Episode 3
// ----------------------------------------------------------------------
// Protocol:			English (GMO)
// ----------------------------------------------------------------------
// Developers:			zTeam / X-MU Community
// ----------------------------------------------------------------------
// Start Date:			20.01.2013
// ----------------------------------------------------------------------
// Update version:		1.0.0.99
// ======================================================================

[FixMe]
1> Need delete old period system and include it to periodex system
1> Need finish CashShop
1> Need add all items effects (cashshop rings and etc)
1> Try Attack In Not Attack Area error when using 265 (Dragon Slasher) & 262 (Chain Drive) skills
1> Need check all calls to gObjRemoveBuffEffect
1> ResistanceCheck -> decompile

[20.01.2013]
1> [stafx.h] Added define 'SEASON6DOT3_ENG'
1> Deleted SEASON5DOT2_JPN, all changes undefined
1> Changed XOR32 keys
1> More test changes in DSProtocol, user and etc
1> [JewelMixSystem.cpp] Fixed item delete in ::MixJewel
1> [JewelMixSystem.cpp] New mix system finished, 99% decompiled from 6.2 THA

[21.01.2013]
1> Finished 'ExpandedInventory' and 'ExpandedWarehouse' systems, defined in <EXPINV>

[22.01.2013]
1> [zzzitem.h] define <MAX_TYPE_PLAYER> changed from 6 to 7, added MONK

1> [zzzitem.cpp] <OpenItemScript> Updated, added MONK class read for item script
1> [protocol.cpp] <PChatProc> Gens chat added, decompiled from 6.2 THA, but without exdb function
1> Deleted ExpandedInventory' and 'ExpandedWarehouse' systems, becoz are bugged, need start from null again

[23.01.2013]
1> [ItemSeedSphere.cpp] Fully decompiled from 1.01.00, all names are original

[27.01.2013]
1> [GameMain.cpp] <GameMainInit> Changed map loading to 1.01.00
1> Max magic size changed to 150 (450) by 1.01.00
1> Many changes with some skills check for RF

[29.01.2013]
1> [ObjUseSkill.cpp] Added 7 new functions to class, <SkillCharge> & <SkillMonkBuff> fully decompiled from 1.01.00
1> [ObjUseSkill.cpp] <EnableSkill> changed, added RF skills (no master>
1> [ObjUseSkill.cpp] <UseSkill> changed, added new cases for RF & doppelganger (1.01.00)
1> [ObjUseSkill.cpp] <RunningSkill> changed, added new casses for RF (1.01.00)
1> [ObjUseSkill.cpp] Changed 1 function name to original (SkillAreaMonsterAttack)
1> [ObjUseSkill.cpp] <SkillEvilMonster> (90%), <SkillMonkBarrageJustOneTarget> (100%) decompiled
1> [ObjAttack.cpp] <Attack> Changed (+2 new arguments) by 1.01.00
1> [protocol.cpp] Added new cases 0x4A & 0x4B for RF attacks
1> [protocol.cpp] Added 3 new functions <GCMonkMagicAttack> (0%), <GCMonkDarkSideTargetSelect> (60%), <GCMonkMagicAttackNumberSend> (95%) by 1.01.00

[06.02.2013]
1> [ObjUseSkill.cpp] Finished <SkillMonkBuffApplyParty>, <SkillMonkBarrageJustOneTarget>, <SkillMonkDarkSideGetTargetIndex> (all decompiled from 1.01.00 by eRRoR, thanks)
1> [Attack.cpp] Finished <CObjAttack::Attack> (decompiled from 1.01.00, but for temp master elements are commented, thanks for eRRoR again)
1> Added "Base" folder by 1.01.00 with some test templates for new libs
1> Added "Lua" folder with lua lib and files for MuLua
1> Added LuaFun / LuaSample / MuLua cpp & h files fully decompiled from 1.01.00
1> Added WinConsole, fully decompiled from 1.01.00
1> [LogProc.cpp] Updated, added some new functions from 1.01.00 (all done)
1> [StdAfx.h] Added lua51.lib load
1> Added MonsterRegenSystem, fully decompiled from 1.01.00
1> Added MonsterStatCalc, fully decompiled from 1.01.00
1> [user.h] Added MASTERLEVEL_PASSIVE_SKILL_OPTION struct (98% decompiled) & defined in OBJECTSTUCT by m_MPSkillOpt
1> [zzzitem.cpp] Fully decompiled from 1.01.00, Period argument on Convert defined in "TEMP_PERIOD_VAR" for temp

[09.02.2013]
1> All BuffSystem decompiled from 1.01.00, all names are original

[12.02.2013]
1> Added lua NpcTalk extension (LuaNPC) + some classes to work with lua scripts

[13.02.2013]
1> Added lua MonsterDie extension (LuaMonsterDie)

[16.02.2013]
1> Expanded inventory fully complete, one by one from 1.01.00 (decompiled)
1> [user.cpp] Added <CheckOutOfWarehouse>, <WarehouseExtentCheck> (fully decompiled from 1.01.00)
1> Expanded warehouse fully complete, one by one from 1.01.00 (decompiled)
1> Added <PeriodItemEx> system, fully decompiled from 1.01.00, need refenses for "TEMP_PERIOD_VAR" define and add data to DB & DS
1> More fixes with RF, all move gates are work normaly for now

[17.02.2013]
1> Added TemporaryUserManager (for now only to PartySystem), 95% decompiled from 1.01.00 (5% for defines on etc files =D)
1> [prodef.h], [protocol.cpp], etc - increased max chat len from 60 to 89 for GMO 6.3 (SEASON6DOT3_ENG)
1> ImperialGuardian / ImperialGuardianRewardExp / EventDungeonItemBag fully finished, decompiled from 1.01.00
1> Fixed all warp / map attr / gate bugs with ImperialGuardian

[18.02.2013]
1> [MasterLevelSystem.cpp] Fully decompiled from 1.01.00 (All names are original), need include
1> [MasterLevelSystem.cpp] Finished, but need uncomment some calls after complete SkillSystem and etc
1> [ItemBag.cpp] Compared with 1.01.00 - All OK

[22.02.2013]
1> [ProbabilityItemBag.cpp] Complete for 1.01.00, but need test, bcz don't sure on 100%
1> [protocol.cpp] <CGItemDropRequest> Complete for 1.01.00, but need test too)
1> [Event.cpp] Added 6 new functions from 1.01.00 for new itembags
1> [GameMain.cpp] <LoadItemBag> Added new boxes, but need recheck all after luckysystem and etc. blya
1> [Event.cpp] <GMPresentBoxItemBagOpen> Fixed box open (X / Y)
1> [GensSystem.cpp] <CalGensClass> Fixed class calculation for Super Knight
1> [GensSystem.cpp] Gens Reward system are finished
1> Monk wings & mini wings complete (exception: 005A43CB, 00665F52)
1> [GensSystem] Remaded RankingUpdater, included to zDataServer timer

zClient_X/zClient_S6E3/Defines.h
#pragma once
// ----------------------------------------------------------------------------------------------

#define ITEM_INTER					1171
#define ITEM(x, y)					((x * 512) + y)
#define ITEM2(x, y)					((x * 512) + y + ITEM_INTER)
#define Color4f(r, b, g, o)			((o << 24) + (g << 16) + (b << 8) + r)
#define Naked(Function)				void __declspec(naked) Function()
#define _FC							__fastcall
#define _CD							__cdecl
#define IS_NEWWINGS(x)				((x < ITEM(12, 180)) ? FALSE : (x > ITEM(12, 186)) ? FALSE : TRUE)
#define CS_GET_WEAPON(x, y)			( y | (16 * (x & 0xF0)) )
#define CS_GET_HELM(x, y, z)		( (32 * (z & 0xF)) + 16 * ((x >> 7) & 1) + (y >> 4) )
#define CS_GET_ARMOR(x, y, z)		( (32 * ((z >> 4) & 0xF)) + 16 * ((x >> 6) & 1) + (y & 0xF) )
#define CS_GET_PANTS(x, y, z)		( (32 * (z & 0xF)) + 16 * ((x >> 5) & 1) + (y >> 4) )
#define CS_GET_GLOVES(x, y, z)		( (32 * ((z >> 4) & 0xF)) + 16 * ((x >> 4) & 1) + (y & 0xF) )
#define CS_GET_BOOTS(x, y, z)		( (32 * (z & 0xF)) + 16 * ((x >> 3) & 1) + (y >> 4) )
#define MAX_ITEMOPT_BIT				0x3F
#define GET_ITEMOPT_LEVEL(x)		((x >> 3) & 0xF)
#define GET_ITEMOPT0(x)				(x & 3)
#define GET_ITEMOPT16(x)			(x & 0x40)
#define GET_ITEMOPT_SKILL(x)		(x >> 7)
#define GET_ITEMOPT_LUCK(x)			((x >> 2) & 1)
#define GET_ITEMOPT_0(x)			(x & 1)
#define GET_ITEMOPT_1(x)			(((x & MAX_ITEMOPT_BIT) >> 1) & 1)
#define GET_ITEMOPT_2(x)			(((x & MAX_ITEMOPT_BIT) >> 2) & 1)
#define GET_ITEMOPT_3(x)			(((x & MAX_ITEMOPT_BIT) >> 3) & 1)
#define GET_ITEMOPT_4(x)			(((x & MAX_ITEMOPT_BIT) >> 4) & 1)
#define GET_ITEMOPT_5(x)			(((x & MAX_ITEMOPT_BIT) >> 5) & 1)
// ----------------------------------------------------------------------------------------------

enum eObjectType	//-> Complete
{
	emPlayer	= 1,
	emMonster	= 2,
	emNPC		= 4,
	emNone		= 8,
};
// ----------------------------------------------------------------------------------------------

enum eClassNumber 
{
	CLASS_WIZARD	= 0,
	CLASS_KNIGHT	= 1,
	CLASS_ELF		= 2,
	CLASS_MAGUMSA	= 3,
	CLASS_DARKLORD	= 4,
	CLASS_SUMMONER	= 5,
	CLASS_MONK		= 6,
};
// ----------------------------------------------------------------------------------------------

enum eClassCode // -> Complete -> Season 5.2 JPN
{
	DarkWizard 		= 0,
	SoulMaster 		= 8,
	GrandMaster		= 24,
	// ----
	DarkKnight		= 1,
	BladeKnight		= 9,
	BladeMaster		= 25,
	// ----
	Elf				= 2,
	MuseElf			= 10,
	HightElf		= 26,
	// ----
	MagicGladiator	= 3,
	DuelMaster		= 19,
	// ----
	DarkLord		= 4,
	LordEmperor		= 20,
	// ----
	Summoner		= 5,
	BloodySummoner	= 13,
	DimensionMaster	= 29,
	// ----
	Monk			= 6,
	FirstMaster		= 16
};
// ----------------------------------------------------------------------------------------------

enum ObjState // -> Complete
{
	SelectServer	= 2,
	SwitchCharacter = 4,
	GameProcess		= 5,
};
// ----------------------------------------------------------------------------------------------

enum ObjWindow // -> Season 6.3
{
	FriendList		= 1,
	MoveList		= 2,
	Party			= 3,
	Guild			= 6,
	Trade			= 7,
	Warehouse		= 8,
	ChaosBox		= 9,
	CommandWindow	= 10,
	PetInfo			= 11,
	Shop			= 12,
	Inventory		= 13,
	Store			= 14,
	OtherStore		= 15,
	Character		= 16,
	ChatWindow		= 33,
	FastMenu		= 34,
	Options			= 35,
	Help			= 36,
	FastDial		= 39,
	SkillTree		= 57,
	NPC_Titus		= 63,
	CashShop		= 65,
	FullMap			= 72,
	NPC_Dialog		= 73,
	GensInfo		= 74,
	NPC_Julia		= 75,
	ExpandInventory	= 77,
	ExpandWarehouse	= 78,
};
// ----------------------------------------------------------------------------------------------

namespace ASM
{
	enum T
	{
		JMP			= 0xE9,
		JE			= 0x74,
		JNE			= 0x75,
		JGE			= 0x7D,
		NOP			= 0x90,
		CALL		= 0xE8,
		PUSH		= 0x68,
	};
};
// ----------------------------------------------------------------------------------------------

enum eTextColors	//-> Complete
{
	White			= 0x00,
	Blue			= 0x01,
	Red				= 0x02,
	Gold			= 0x03,
	Green			= 0x04,
	WhiteRed		= 0x05,
	Violet			= 0x06,
	WhiteBlue		= 0x07,
	WhiteBeige		= 0x08,
	GreenBlue		= 0x09,
	Gray			= 0x0A,
	DarkViolet		= 0x0B,
};
// ----------------------------------------------------------------------------------------------

enum eTextWeight	//-> Complete
{
	Bold			= 0x01,
	Normal			= 0x00,
};
// ----------------------------------------------------------------------------------------------

namespace eItemOption
{
	enum T
	{
		Damage			= 80,		//-> Additional Damage +<N>
		WizardryDamage	= 81,
		CurseSpell		= 82,
		Defense			= 84,		//-> Additional Defense +<N>
		Luck			= 85,		//-> Default: 0
		HPRecovery		= 86,
		// ----
		LifeIncrease	= 87,		//-> Increase Max HP +4%%
		ManaIncrease	= 88,		//-> Increase Max Mana +4%%
		DamageDecrease	= 89,		//-> Damage Decrease +4%%
		Reflect			= 90,		//-> Reflect Damage +5%%
		DefenseSuccess	= 91,		//-> Defense success rate +10%%
		IncreaseZen		= 92,		//-> Increases acquisition rate of Zen after hunting monsters +30%%
		// ----
		ExcellentDamage	= 93,		//-> Excellent Damage rate +10%%
		DamageByLevel	= 94,		//-> Increase Damage +level/20
		IncreaseDamage	= 95,		//-> Increase Damage +%d%%
		IncreaseWizardry= 96,		//-> Increase Wizardry Dmg +level/20
		IncreaseWizardryDmg = 97,	//-> Increase Wizardry Dmg +2%
		IncreaseSpeed	= 98,		//-> Increase Attacking(Wizardry)speed +%d
		LifePerMob		= 99,		//-> Increases acquisition rate of Life after hunting monsters +life/8
		ManaPerMob		= 100,		//-> Increases acquisition rate of Mana after hunting monsters +Mana/8
		IgnorOpDefense	= 103,
		ReturnAttack	= 106,		//-> Return's the enemy's attack power in <N>%
		CompleteLife	= 107,		//-> Complete recovery of life in <N>% rate
		CompleteMana	= 108,		//-> Complete recovery of Mana in <N>% rate
	};
};
// ----------------------------------------------------------------------------------------------